### PR TITLE
:arrow_up::one::two:, pos_keyboard:

### PR DESCRIPTION
**Pull request by fedoranvar**
Refactoring

### CODE REVIEW

File: pos_keyboard/__manifest__.py [hunk 1]
```diff
@@ -3,14 +3,15 @@
...
+# Copyright 2019 Kildebekov Anvar <https://it-projects.info/team/kildebekov>
 # License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
 {
     "name": """Keyboard support in Point Of Sale""",
     "summary": """Module allows to use usual keyboard (not virtual one) in Point of Sale""",
     "category": "Point Of Sale",
     # "live_test_url": "http://apps.it-projects.info/shop/product/DEMO-URL?version={ODOO_BRANCH}",
     "images": ['images/keyboard.png'],
-    "version": "11.0.1.1.3",
+    "version": "12.0.1.0.0",
```
Comment (KolushovAlexandr) [inline pos_keyboard/__manifest__.py]: do not change the version when porting, especially downgrading

File: pos_keyboard/doc/changelog.rst [hunk 2]
```diff
@@ -1,30 +1,4 @@
-`1.1.3`
+`1.0.0`
```
Comment (KolushovAlexandr) [inline pos_keyboard/doc/changelog.rst]: revert those changes

File: pos_keyboard/README.rst [hunk 3]
```diff
@@ -41,4 +42,4 @@ Usage instructions: `<doc/index.rst>`_
 
 Changelog: `<doc/changelog.rst>`_
 
-Tested on Odoo 11.0 77e7d952b4c72d58c1b8f34b57e355f5b98c88b0
+Tested on Odoo 12.0 34369ff58121b33387aada0929e34efe783c3baf
```
Comment (KolushovAlexandr) [inline pos_keyboard/README.rst]: change all references to 11.0v here and probably in index.rst

**Review by KolushovAlexandr — CHANGES_REQUESTED** [review]

File: pos_keyboard/__manifest__.py (hunk 1 above)
Comment (fedoranvar) [reply pos_keyboard/__manifest__.py]: why porting from 11 to 12 - is downgrading?

Comment (fedoranvar) [reply pos_keyboard/__manifest__.py]: > do not change the version when porting, especially downgrading

why porting from 11 to 12 - is downgrading?

File: pos_keyboard/README.rst (hunk 3 above)
Comment (fedoranvar) [reply pos_keyboard/README.rst]: > change all references to 11.0v here and probably in index.rst

why?

File: pos_keyboard/__manifest__.py (hunk 1 above)
Comment (KolushovAlexandr) [reply pos_keyboard/__manifest__.py]: you downgraded modules version 1.1.3 to 1.0.0

File: pos_keyboard/README.rst (hunk 3 above)
Comment (KolushovAlexandr) [reply pos_keyboard/README.rst]: sorry, meant to 12.0

Comment (KolushovAlexandr) [reply pos_keyboard/README.rst]: but anyway you understood me :+1:

File: pos_keyboard/__manifest__.py (hunk 1 above)
Comment (fedoranvar) [reply pos_keyboard/__manifest__.py]: Resolved

**Review by KolushovAlexandr — CHANGES_REQUESTED** [review]
@fedoranvar When I click number keys in the password popup they are added doubled

**Review by KolushovAlexandr — APPROVED** [review]

**Review by Ramil-Mukhametzyanov — COMMENTED** [review]
There is reference to Odoo 10.0 in the static/description/index.html file.

File: pos_keyboard/static/description/index.html [hunk 4]
```diff
@@ -62,7 +62,7 @@ <h2>Need our service?</h2>
                             -o-transform: rotate(8deg);
                             -moz-transform: rotate(8deg);
                             -ms-transform: rotate(8deg);">
-                    Tested on Odoo<br/>10.0 community
+                    Tested on Odoo<br/>11.0 community
```
Comment (Ramil-Mukhametzyanov) [inline pos_keyboard/static/description/index.html]: @fedoranvar  Porting to Odoo 12.0

**Review by Ramil-Mukhametzyanov — APPROVED** [review]

**Review by KolushovAlexandr — APPROVED** [review]

**Review by em230418 — CHANGES_REQUESTED** [review]
Scenario:
- login as demo
- Point of Sale -> New session
- Click on any item in shop
- Click on "Payment"
- Typing in usual keyboard, for example 987, nothing appears. Expecting something to be appeared.
- Clicking on any button in virtual keyboard. For example 1. Finally something appears with previous "987".

**Comment by fedoranvar** [comment]
> Scenario:
> 
> * login as demo
> * Point of Sale -> New session
> * Click on any item in shop
> * Click on "Payment"
> * Typing in usual keyboard, for example 987, nothing appears. Expecting something to be appeared.
> * Clicking on any button in virtual keyboard. For example 1. Finally something appears with previous "987".

Эта проблема работы Оду, не работы этого модуля.
проверялось на ранботе оду.

**Comment by em230418** [comment]
Давай заодно и это с тестами: https://github.com/it-projects-llc/pos-addons/pull/1084/commits/b30d0a55dcd6dc5ec8745866aff4ac9c7ee53cd5

**Review by em230418 — APPROVED** [review]

**Comment by em230418** [comment]
I approve to merge it now

File: pos_keyboard/doc/changelog.rst [hunk 5]
```diff
@@ -1,3 +1,8 @@
+`1.1.4`
+-------
+
+**Fix:** Incompatibility with some modules led to the action duplicating
```
Comment (Ramil-Mukhametzyanov) [inline pos_keyboard/doc/changelog.rst:10]: :+1: checked

**Pull request merged**